maps/maps.go

```diff
@@ -254,25 +255,31 @@ func IntfaceKeysToStrings(mp map[string]interface{}) {
 			mp[key] = x
 			IntfaceKeysToStrings(x)
 		case []interface{}:
-			for i, v := range cur {
-				switch sub := v.(type) {
-				case map[interface{}]interface{}:
-					x := make(map[string]interface{})
-					for k, v := range sub {
-						x[fmt.Sprintf("%v", k)] = v
-					}
-					cur[i] = x
-					IntfaceKeysToStrings(x)
-				case map[string]interface{}:
-					IntfaceKeysToStrings(sub)
-				}
-			}
+			sliceIntfaceKeysToStrings(cur)
 		case map[string]interface{}:
 			IntfaceKeysToStrings(cur)
 		}
 	}
 }
 
+func sliceIntfaceKeysToStrings(cur []interface{}) {
```

Comment:
There is a small caveat, if someone does add a "circular dependency" (slice has a reference to itself) then this will get into an infinite loop, not sure if anything to worry.